queue/doubleendedqueue.js

```diff
@@ -0,0 +1,117 @@
+
+class DoubleEndedQueue {
+    constructor (maxCapacity = 10) {
```

Comment:
can we use following names for operations which makes more sense https://www.geeksforgeeks.org/deque-set-1-introduction-applications/
Operations on Deque:
Mainly the following four basic operations are performed on queue:

insertFront(): Adds an item at the front of Deque.
insertLast(): Adds an item at the rear of Deque.
deleteFront(): Deletes an item from front of Deque.
deleteLast(): Deletes an item from rear of Deque.

In addition to above operations, following operations are also supported
getFront(): Gets the front item from queue.
getRear(): Gets the last item from queue.
isEmpty(): Checks whether Deque is empty or not.
isFull(): Checks whether Deque is full or not.